Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -172,3 +190,81 @@ function IsFilePathSpecified
                 [StringComparison]::OrdinalIgnoreCase)
 }
 
+
+function GetVersion($uri)
+{
+    $version = $null
+
+    Try
+    {
+        $version = Invoke-RestMethod -Uri $uri -Method Get
+    }
+    Catch [System.Net.WebException]
+    {
+        Write-Verbose "WebException while trying to invoke $uri. Exception msg:$($_.Exception.Message)"
+    }
+
+    return $version
+}
+
+#
+# Helper that returns the version number of the SonarQube server
+#
+function GetSonarQubeServerVersion()
+{
+    param([String][ValidateNotNullOrEmpty()]$serverUrl)
+
+    Write-Host "Fetching SonarQube server version.."
+
+    $serverUri = New-Object -TypeName System.Uri -ArgumentList $serverUrl
+    $serverApiUri = New-Object -TypeName System.Uri -ArgumentList ($serverUri, "/api/server/version")
+
+    $sqVersion = GetVersion $serverApiUri
+
+    if(!$sqVersion)
+    {
+        Write-Verbose "Trying to fetch SonarQube version number again.."
+        Start-Sleep -s 2
+
+        $sqVersion = GetVersion $serverApiUri
+    }
+
+    Write-Verbose "Returning SonarQube server version:$sqVersion"
+    return $sqVersion
+}
+
+#
+# Helper that returns the major version number of the SonarQube server
+#
+function GetSQMajorVersionNumber()
+{
+    param([String]$sqServerVersion)
+
+    [int]$majorVersion = 0;
+    $tokens = $sqServerVersion.Split(".")
+
+    if ($tokens -and $tokens.Count -ge 1)
+    {
+        $result = [int]::TryParse($tokens[0], [ref]$majorVersion)
+    }
+
+    return $majorVersion
+}
+
+#
+# Helper that returns the minor version number of the SonarQube server
+#
+function GetSQMinorVersionNumber()
```

Comment:
There is quite a bit of code around parsing the version and comparing it. Could you look into using the .net Version class instead? https://msdn.microsoft.com/en-us/library/system.version(v=vs.110).aspx